toolchain/check/context.cpp

```diff
@@ -167,6 +167,22 @@ auto Context::NoteIncompleteClass(SemIR::ClassId class_id,
   }
 }
 
+auto Context::NoteUndefinedInterface(SemIR::InterfaceId interface_id,
+                                     DiagnosticBuilder& builder) -> void {
+  CARBON_DIAGNOSTIC(InterfaceForwardDeclaredHere, Note,
+                    "Interface was forward declared here.");
+  CARBON_DIAGNOSTIC(InterfaceUndefinedWithinDefinition, Note,
+                    "Interface is currently being defined.");
```

Comment:
Maybe move these into the respective `if` statements.

Reply:
Done.